src/openzaak/components/documenten/models.py

```diff
@@ -466,6 +468,20 @@ def locked(self, value: bool) -> None:
         self._locked = value
 
     def save(self, *args, **kwargs) -> None:
+        is_new_instance = self.pk is None
```

Comment:
`EnkelvoudigInformatieObject` inherits from `InformatieObject` and since the identification is currently generated in `InformatieObject.save`, let's move the changes made in `EnkelvoudigInformatieObject.save` to `InformatieObject.save`.

Let's also add the decorator `@transaction.atomic` to make sure the deletion of ReservedDocuments is rolled back if anything fails while saving the `EnkelvoudigInformatieObject`

```
from django.db import transaction

@transaction.atomic
def save(...):
    ...
```